WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,18 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    self.rounds.each(&:destroy!)
```

Comment:
Unless I misunderstand it, `destroy_all` doesn't throw an exception if it fails to delete an object, and there is no `destroy_all!`.